cpp/src/binaryop/binaryop.cpp

```diff
@@ -306,7 +307,56 @@ std::unique_ptr<column> binary_operation(column_view const& lhs,
       output_type, lhs.size(), std::move(new_mask), cudf::UNKNOWN_NULL_COUNT, stream, mr);
   }
 
-  if (lhs.size() == 0) { return out; }
+  if (lhs.is_empty()) return out;
+
+  auto out_view = out->mutable_view();
+  binops::jit::binary_operation(out_view, lhs, rhs, op, stream);
+  return out;
+}
+
+std::unique_ptr<column> make_fixed_width_column_for_output(column_view const& lhs,
+                                                           column_view const& rhs,
+                                                           binary_operator op,
+                                                           data_type output_type,
+                                                           rmm::mr::device_memory_resource* mr,
+                                                           cudaStream_t stream)
+{
+  if (binops::null_using_binop(op)) {
+    return make_fixed_width_column(output_type, rhs.size(), mask_state::ALL_VALID, stream, mr);
+  } else {
+    auto new_mask = bitmask_and(table_view({lhs, rhs}), mr, stream);
+    return make_fixed_width_column(
+      output_type, lhs.size(), std::move(new_mask), cudf::UNKNOWN_NULL_COUNT, stream, mr);
+  }
+};
+
+int32_t apply_scale_binop(binary_operator op, int32_t left_scale, int32_t right_scale)
+{
+  if (op == binary_operator::MUL) return left_scale + right_scale;
+  if (op == binary_operator::DIV) return left_scale - right_scale;
+  return std::min(left_scale, right_scale);
+}
+
+std::unique_ptr<column> fixed_point_binary_operation(column_view const& lhs,
+                                                     column_view const& rhs,
+                                                     binary_operator op,
+                                                     rmm::mr::device_memory_resource* mr,
+                                                     cudaStream_t stream)
+{
+  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(),
+               "Both columns must be of the same fixed_point type");
+
+  auto const scale       = apply_scale_binop(op, lhs.type().scale(), rhs.type().scale());
+  auto const output_type = data_type{lhs.type().id(), scale};
+  auto out = make_fixed_width_column_for_output(lhs, rhs, op, output_type, mr, stream);
+
+  // Check for 0 sized data
+  if (lhs.is_empty() || rhs.is_empty()) return out;
+
+  // Adjust columns so they have they same scale
+  if (lhs.type().scale() != rhs.type().scale()) {
+    // TODO
```

Comment:
Oh this is a big todo!

Reply:
Yes, the is the last thing I am wrapping up.

Reply:
Done.